### PR TITLE
Include Compatibility matrix and update the readme

### LINKED ISSUE #359: Document the compatibility metrics
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
e.g: AWS metrics https://github.com/kubernetes-sigs/cluster-api-provider-aws#compatibility-with-cluster-api-and-kubernetes-versions

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:

- Added Compatibility matrix
- Removed all the unnecessary content and mentioned the hosted doc for the references

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #359 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,204 +14,31 @@ This repository hosts a concrete implementation of an IBM Cloud provider for the
...
-    ```
+## Documentation
 
-    Kubeadm Control Plane
-    ```console
-    ~ kubectl get kubeadmcontrolplane
-    NAME                       INITIALIZED   API SERVER AVAILABLE   VERSION   REPLICAS   READY   UPDATED   UNAVAILABLE
-    ibm-vpc-0-control-plane    true          true                   v1.19.9   1          1       1
-    ```
+Please see our [book](https://cluster-api-ibmcloud.sigs.k8s.io) for in-depth documentation.
```
Comment (gyliu513) [inline README.md:19]: The book looks great, thanks @mkumatag !

File: README.md [hunk 2]
```diff
@@ -2,23 +2,6 @@
 [![Releases](https://img.shields.io/badge/version-v0.0.1-orange.svg)](https://sigs.k8s.io/cluster-api-provider-ibmcloud/releases "Cluster API provider IBM Cloud latest release")
 [![Go Report Card](https://goreportcard.com/badge/sigs.k8s.io/cluster-api-provider-ibmcloud)](https://goreportcard.com/report/sigs.k8s.io/cluster-api-provider-ibmcloud)
 
-<!-- START doctoc generated TOC please keep comment here to allow auto update -->
```
Comment (gyliu513) [inline README.md:5]: DO we still want to keep the toc? Why remove it?

Comment (mkumatag) [reply README.md:5]: we hardly have much content here in the readme to maintain the index, wdys?

**Pull request merged**